harvesting/esciencecenter.py

```diff
@@ -0,0 +1,469 @@
+    else: 
+        logger.warning(f"failed storing person in database: {name}")
+
+
+def get_suffix(url):
+    url_parts = url.strip().split(".")
+    if len(url_parts) > 0: return(url_parts[-1])
+    else: return("")
+
+
+def make_new_organization_data(name_data):
+    new_organization_data = {"primaryKey":{"id":str(uuid.uuid4()),"collection":"organization"}}
+    new_organization_data["createdAt"] = get_current_time_string()
+    new_organization_data["createdBy"] = "esciencecenter.nl.crawler"
+    new_organization_data["updatedAt"] = new_organization_data["createdAt"]
+    new_organization_data["updatedBy"] = new_organization_data["createdBy"]
+    new_organization_data["url"] = "http://fix.me"
+    if "name" in name_data: new_organization_data["name"] = name_data["name"]
+    img_data = ""
+    img_suffix = "png"
+    if "img_url" in name_data:
+        img_response = requests.get(name_data["img_url"])
+        if request_was_successful(img_response): 
+            img_data = base64.b64encode(img_response.content).decode("utf-8") 
+            img_suffix = get_suffix(name_data["img_url"])
+    new_organization_data["logo"] = {"data":img_data, "mimeType":"image/"+img_suffix}
+    return(new_organization_data)
+
+
+def add_new_organization_to_database(token,api_organizations_data,name_data):
+    new_organization_data = make_new_organization_data(name_data)
+    organization_api_url = "https://localhost/api/organization"
+    organization_api_response = requests.post(organization_api_url,json=new_organization_data,headers={'Authorization': 'Bearer %s' % token},verify=False)
+    if request_was_successful(organization_api_response):
+        api_organizations_data.append(new_organization_data)
+        print(f'added new organization "{name_data["name"]}"')
+        return(new_organization_data)
+    else: 
+        logger.warning(f"failed storing organization in database: {name}")
+        
+
+# see required list and patterns in backend/schemas/project.json
+def make_new_project_data(project_data_source):
+    new_project_data = {}
+    new_project_data["callUrl"] = "https://doi.org/fix.me"
+    new_project_data["codeUrl"] = "https://github.com/fix.me"
+    new_project_data["createdAt"] = get_current_time_string()
+    new_project_data["createdBy"] = "esciencecenter.nl.crawler"
+    new_project_data["dateEnd"] = "1900-12-31T23:59:59Z"
+    new_project_data["dateStart"] = "1900-01-01T00:00:00Z"
+    new_project_data["description"] = remove_html_links(project_data_source["text"])
+    new_project_data["grantId"] = "FIX ME"
+    new_project_data["imageCaption"] = "FIX ME FIX ME"
+    new_project_data["imageUrl"] = project_data_source["imageUrl"]
+    new_project_data["impact"] = []
+    new_project_data["isPublished"] = True
+    new_project_data["output"] = []
+    new_project_data["primaryKey"] = {"id":str(uuid.uuid4()),"collection":"project"}
+    new_project_data["related"] = {"organizations":[],"projects":[],"software":[]}
```

Comment:
Why are related projects & orgs, team not filled for new projects, while they are filled in update_project_data() for existing project?

Reply:
Nevermind just tested it and project/orgs/team is filled.

Reply:
I did not want to duplicate that part of the code. My strategy for new projects is to fill a basic template for the project (line 449)  and then run the existing project update code on this template (line 462), which adds the related projects and organizations.